addons/web_editor/controllers/main.py

```diff
@@ -530,6 +530,16 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
             fields['res_id'] = res_id
         if name:
             fields['name'] = name
+
+        # Restricted editors can handle attachments related to records to which they have access
+        # Would user be able to read fields of original record ?
+        if attachment and attachment.res_model and attachment.res_id:
+            request.env[attachment.res_model].browse(attachment.res_id).with_user(request.env.user).check_access_rights('read')
+
+        # Would user be able to write fields of target record ?
+        # Rights check works with res_id=0 because browse(0) returns an empty record set
+        request.env[fields['res_model']].browse(fields['res_id']).with_user(request.env.user).check_access_rights('write')
```

Comment:
Is it safe to rely on `fields['res_model']` ? The user could lie on that no ? It's just a param given in the request.

Reply:
Sure, but that's where the information is written: if the user lies on `res_model` by making it a blog instead of a product, the check has to be done on blog because it is a blog record that would be modified if the operation goes through.
So, IMO, this is ok.

Reply:
My fear is that a user with right on blog post could modify image of product by calling modify_image with the ID of an attachment linked to a product but in the request param he pass blog.post as res_model.

Reply:
`modify_image` creates attachments, it does not actually modify the original - it only keeps the link to the original in `original_id`.